packages/core/src/agent/didcomm/versions/v2/DIDCommV2BaseMessage.ts

```diff
@@ -0,0 +1,164 @@
+import type { ParsedMessageType } from '../../../../utils/messageType'
+import type { Constructor } from '../../../../utils/mixins'
+import type { DIDCommV1BaseMessage } from '../v1/DIDCommV1BaseMessage'
+
+import { Expose } from 'class-transformer'
+import { IsArray, IsNumber, IsOptional, IsString, Matches } from 'class-validator'
+
+import { JsonEncoder } from '../../../../utils'
+import { uuid } from '../../../../utils/uuid'
+import { MessageIdRegExp, MessageTypeRegExp } from '../../validation'
+
+export const ATTACHMENT_MEDIA_TYPE = 'application/json'
+
+export type DIDComV2BaseMessageConstructor = Constructor<DIDCommV1BaseMessage>
+
+export type DIDCommV2MessageParams = {
+  type?: string
+  id?: string
+  from?: string
+  to?: string | string[]
+  thid?: string
+  pthid?: string
+  created_time?: number
+  expires_time?: number
+  from_prior?: string
+  attachments?: Array<Attachment>
+  body?: unknown
+}
+
+export class DIDCommV2BaseMessage {
+  @Expose({ name: 'id' })
+  @Matches(MessageIdRegExp)
+  public id!: string
+
+  @Expose({ name: 'type' })
+  @Matches(MessageTypeRegExp)
+  public readonly type!: string
+  public static readonly type: ParsedMessageType
+
+  @Expose({ name: 'typ' })
+  public readonly typ = DIDCommV2BaseMessage.typ
+  public static readonly typ = 'application/didcomm-plain+json'
```

Comment:
Can we make this a full word (is is mimeType? I think it's called different today, mediaType?)